api/user/userController.js

```diff
@@ -1,6 +1,76 @@
 'use strict';
 
+var User = require('./userModel').User;
+var Friend = require('./userModel').Friend;
+
 exports.create = function (req, res) {
 
-   res.send("sad");
+    var friend = new Friend( { name: req.body.name });
+
+    var user = new User({
+        name: req.body.name,
+        friends: [friend]}
+    );
+
+    user.save(function (err,user) {
+        if (err) return handleError(err);
+         res.send(user);
+    });
+};
+
+exports.findUser = function (req, res) {
+
+    User.findById(req.params.id, function (err,user) {
```

Comment:
don't use parameter itself inside find method. Use instead 
let userId = req.params.id; (should self-described name)
User.findById(userId)